src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -776,7 +776,16 @@ internal ClassDataContractCriticalHelper(Type type) : base(type)
                 this.IsValueType = type.IsValueType;
                 if (baseType != null && baseType != Globals.TypeOfObject && baseType != Globals.TypeOfValueType && baseType != Globals.TypeOfUri)
                 {
+#if uapaot
+                    // The DataContact pre-generated by SG is imcomplete, in other words, some of its fields are not
+                    // populated by SG. Reflection-based serialization, however, requires a complete DataContract.
+                    // At this point, we're in the middle of creating a new DataContract at runtime, which indicates
+                    // reflection-based serialization is being used. Therefore, we cannot use the pre-generated DataContracts
+                    // of base types, even if they exist. 
+                    DataContract baseContract = DataContract.GetDataContractCreatedAtRuntime(baseType);
```

Comment:
Just curious: So this means that when using reflection based serializer we will always use the runtime created data contract, right? I mean it makes sense and it simplifies the code (no need to deal with mismatches between the data contracts).

That said - what happens if I use SG on CoreCLR? Or do we not support it there?